### PR TITLE
Enhancement/72 remove ossechids conditional directives from the code

### PR DESCRIPTION
**Pull request by sdvendramini**
Related issue:

- https://github.com/wazuh/wazuh-agent/issues/72

## Description

This PR adds the changes to delete the use of the OSSECHIDS conditional directive and remove any unnecessary code used along with these directives.

### CODE REVIEW

File: src/modules/rootcheck/src/check_rc_pids.c [hunk 1]
```diff
@@ -169,13 +169,11 @@ static void loop_all_pids(const char *ps, pid_t max_pid, int *_errors, int *_tot
         }
 
         /* If we are run in the context of OSSEC-HIDS, sleep here (no rush) */
```
Comment (TomasTurina) [inline src/modules/rootcheck/src/check_rc_pids.c:171]: Remove comment.

File: src/modules/rootcheck/src/run_rk_check.c [hunk 2]
```diff
@@ -38,7 +38,6 @@ int notify_rk(int rk_type, const char *msg)
         return (0);
     }
 
-#ifdef OSSECHIDS
     /* When running in context of OSSEC-HIDS, send problem to the rootcheck queue */
```
Comment (TomasTurina) [inline src/modules/rootcheck/src/run_rk_check.c:41]: Remove comment.

File: src/modules/rootcheck/src/rootcheck.c [hunk 3]
```diff
@@ -115,46 +77,6 @@ int rootcheck_init(int test_config)
...
-                break;
-            case 'r':
-                rootcheck.readall = 1;
-                break;
-            default:
-                help_rootcheck(home_path);
-                break;
-        }
-    }
 #ifdef WIN32
```
Comment (TomasTurina) [inline src/modules/rootcheck/src/rootcheck.c:80]: Remove also this block.

File: src/modules/rootcheck/src/check_open_ports.c
Comment (TomasTurina) [inline src/modules/rootcheck/src/check_open_ports.c:1]: If they work, I think we can keep these functions. Please test them, for example in version 4.9.0 (remove the `ifdef` so they compile), and we will see what to do.

File: src/modules/rootcheck/src/check_rc_pids.c (hunk 1 above)
Comment (sdvendramini) [reply src/modules/rootcheck/src/check_rc_pids.c]: Done!

File: src/modules/rootcheck/src/run_rk_check.c (hunk 2 above)
Comment (sdvendramini) [reply src/modules/rootcheck/src/run_rk_check.c]: Done!

File: src/modules/rootcheck/src/rootcheck.c (hunk 3 above)
Comment (sdvendramini) [reply src/modules/rootcheck/src/rootcheck.c]: Done!

File: src/modules/rootcheck/src/check_open_ports.c
Comment (sdvendramini) [reply src/modules/rootcheck/src/check_open_ports.c:1]: I tested with version `4.9.0`, and the port check doesn't do anything, it just prints a message with `printf`. This check is performed according to the default agent configuration setting:
```
<localfile>
    <log_format>full_command</log_format>
    <command>netstat -tulpn | sed 's/\([[:alnum:]]\+\)\ \+[[:digit:]]\+\ \+[[:digit:]]\+\ \+\(.*\):\([[:digit:]]*\)\ \+\([0-9\.\:\*]\+\).\+\ \([[:digit:]]*\/[[:alnum:]\-]*\).*/\1 \2 == \3 == \4 \5/' | sort -k 4 -g | sed 's/ == \(.*\) ==/:\1/' | sed 1,2d</command>
    <alias>netstat listening ports</alias>
    <frequency>360</frequency>
</localfile>
```

**Review by TomasTurina — APPROVED** [review]

**Pull request merged**